src/emqx_protocol.erl

```diff
@@ -873,6 +875,39 @@ parse_topic_filters(?SUBSCRIBE, RawTopicFilters) ->
 parse_topic_filters(?UNSUBSCRIBE, RawTopicFilters) ->
     lists:map(fun emqx_topic:parse/1, RawTopicFilters).
 
+%% Issue#1641: not close connection when http acl rejected
+%%-----------------------------------------------------------------------------
+%% close connection when acl rejected and the config is available
+%%-----------------------------------------------------------------------------
+
+control_acl_pub_connection(PState) ->
+    case is_enable() of
+        true -> shutdown("close connection", PState);
```

Comment:
I don't think the this will disconnect your mqtt client. Have a test and make sure the connection is disconnected.

Reply:
yes! I realize it's mistake

Reply:
yes!  your are right